### PR TITLE
Feature/search

### PR DESCRIPTION
**Pull request by guavajellyaaron**
Added a search feature using jQuery Autocomplete and Dapper.

Also added a services layer for future development.

### CODE REVIEW

File: projects/BossCheatSheets.Services/BossCheatSheets.Services/BossCheatSheets.Services.csproj [hunk 1]
```diff
@@ -0,0 +1,60 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (happygaijin) [inline projects/BossCheatSheets.Services/BossCheatSheets.Services/BossCheatSheets.Services.csproj]: Does this have to be in a subfolder of itself?
BossCheatSheets.Services/BossCheatSheets.Services/BossCheatSheets.Services.csproj

What is this services project for?

File: projects/BossCheatSheets.Web/Controllers/HomeController.cs [hunk 2]
```diff
@@ -47,5 +53,15 @@ public ActionResult StrategyDetail(int id)
...
+        {
+            IDbConnection connection = new SqlConnection(ConfigurationManager.ConnectionStrings["DefaultConnection"].ConnectionString);
+            IEnumerable<SearchViewModel> results = connection.Query<SearchViewModel>("SELECT *,'Boss' AS Type, BossID AS ID FROM Boss WHERE Name LIKE '%" + q + "%'");
+            results = results.Concat(connection.Query<SearchViewModel>("SELECT *,'Instance' AS Type, InstanceID AS ID FROM Instance WHERE Name LIKE '%" + q + "%'"));
+            results = results.Concat(connection.Query<SearchViewModel>("SELECT *,'Expansion' AS Type, ExpansionID AS ID FROM Expansion WHERE Name LIKE '%" + q + "%'"));
+            results = results.Concat(connection.Query<SearchViewModel>("SELECT *,'Strategy' AS Type, StrategyID AS ID, Title AS Name FROM Strategy WHERE Title LIKE '%" + q + "%'"));
+            return Json(results, JsonRequestBehavior.AllowGet);
+        }
     }
 }
```
Comment (happygaijin) [inline projects/BossCheatSheets.Web/Controllers/HomeController.cs]: :+1: 

Just consider the UNION we talked about last night. That way it's one hit to the DB and SQL is faster at concatenating results.

File: projects/BossCheatSheets.Services/BossCheatSheets.Services/BossCheatSheets.Services.csproj (hunk 1 above)
Comment (guavajellyaaron) [inline projects/BossCheatSheets.Services/BossCheatSheets.Services/BossCheatSheets.Services.csproj]: I just followed the existing structure. Is this incorrect?

I'm going to use it for the voting code and other service methods

Comment (happygaijin) [inline projects/BossCheatSheets.Services/BossCheatSheets.Services/BossCheatSheets.Services.csproj]: Yours has "BossCheatSheets.Services" three times. twice for the folder names and once for the project name. The other projects use the name twice. Once for the folder and once for the project file.

**Comment by guavajellyaaron** [comment]
Closing due to same code in feature/custom_routes pull request.

**Pull request closed without merging**